### DISCUSSION TITLE
Python 2 vs Python3

### DISCUSSION THREAD

**Discussion by AngeloIapichino** [miguelgrinberg/python-socketio#817] [Q&A] [answered]
Hi guys,
hope i can have answers and solution.

I have to write a simple script in python2 that print data received from a socketio.

My code (the message received is printed once):
```
import socketio

sio = socketio.Client()
sio.connect('http://192.168.125.100')


@sio.event
def message(data):
    print('Received data: ', data)
    sio.disconnect()
```

The code works well in python 3 installing python-socketio with these lines:
`pip install python-socketio`
`pip install python-socketio[client]`

But with python 2 i cannot find any tips for installing this package and make to working properly.
After some attempts and research i found the combination of python-socketio and python-engineio that don't give me import problems:
`python-engineio==3.11.2`
`python-socketio==4.4.0`
Executing the code in python 2 gives me the error: `socketio.exceptions.ConnectionError: Unexpected status code 400 in server response`

In the Screen below you can see the execution with python2, python3, `pip freeze` and `pip3 freeze`
![onrobot](https://user-images.githubusercontent.com/12069840/139867690-e58e90ed-8e8c-454a-80eb-11cc6e655394.JPG)

**Comment by miguelgrinberg** [comment, marked as answer]
Python 2 has been long unsupported. You have downgraded to old releases of python-socketio and python-engineio, and these older versions implement an older version of the Socket.IO protocol, so you will also need to downgrade your server. See the Version Compatibility chart in the documentation for details. If the server is also written in Python, then this is as simple as using the same versions of these packages in the client and the server.

**Reply by AngeloIapichino** [reply]
Thank you for your answer Miguel. Very kind.

I don't have access to the server, it's a device connected to one robotic gripper and i want to read the realtime data from that.  I saw it uses socketio from his web gui.

Do you know an alternative to your package that support python2 and that version of Socket.IO?

Thank you in advance

**Reply by miguelgrinberg** [reply]
I don't know what version you need, so I'm not sure how I can help. It's going to be probably impossible to find anything up to date for Python 2, so I doubt you will find what you need. Better to upgrade to 3.
